qC.cpp

```diff
@@ -0,0 +1,67 @@
+#include <iostream>
+#include <algorithm>
+#include <vector>
+#include <climits>
+
+using namespace std;
+typedef long long ll;
+
+void update_all_for_jump_n_from(int pos, int jump, ll cost, const vector<ll>& vec, vector<vector<ll>>& min_price_vec){
+    /* debugging prints
+    cout << "pos: " << pos << endl;
+    cout << "jump: " << jump << endl;
+    cout << "cost: " << cost << endl;
+    cout << "-------------" << endl;
+    */
+    jump += 1;
+    if (pos + jump >= vec.size()) return;
+    int new_pos = pos + jump;
+    ll new_cost = cost + vec[new_pos];
+
+    if(new_cost < min_price_vec[new_pos][jump])
+        min_price_vec[new_pos][jump] = new_cost;
+
+    new_pos -= jump;
+    new_cost += vec[new_pos];
+    while (new_pos >= 1){
+        if (new_cost < min_price_vec[new_pos][jump])
+            min_price_vec[new_pos][jump] = new_cost;
+        new_pos -= jump;
+        new_cost +=vec[new_pos];
+    }
+}
+
+
+int main() {
+    int n;
+    while (cin >> n){
```

Comment:
read about istream_iterator to read all inputs automatically:

#include <iterator> //for std::istream_iterator 

std::istream_iterator<int> begin(std::cin), end;
std::vector<int> v(begin, end);

would construct a vector of size n+1 that his first value is n and the rest are the input prices.

Reply:
no need for the WHILE!